packages/cli/src/storage/command.ts

```diff
@@ -0,0 +1,63 @@
+import { logger } from "../core/helpers/logger";
+import { spinner } from "../core/helpers/logger";
+import { ThirdwebStorage } from "@thirdweb-dev/storage";
+import chalk from "chalk";
+import fs from "fs";
+import path from "path";
+
+export async function upload(
+  storage: ThirdwebStorage,
+  uploadPath: string,
+): Promise<string> {
+  const pathExists = fs.existsSync(uploadPath);
+  if (!pathExists) {
+    logger.error(
+      `Invalid path ${chalk.blueBright(
+        uploadPath,
+      )} provided. Please provide a valid path to a file or directory to upload."`,
+    );
+    process.exit(1);
+  }
+
+  let uri = "";
+  const fileType = fs.lstatSync(uploadPath);
+  if (fileType.isDirectory()) {
+    const subPaths = fs.readdirSync(uploadPath);
+    const filePaths = subPaths
+      .map((subPath) => path.join(uploadPath, subPath))
+      .filter((subPath) => fs.lstatSync(subPath).isFile());
+    const files = filePaths.map((filePath) => fs.readFileSync(filePath));
+
+    if (files.length === 0) {
+      logger.error(
+        `No files detected in specified directory ${chalk.blueBright(
+          uploadPath,
+        )} to upload.`,
+      );
+      process.exit(1);
+    }
+
+    const spin = spinner(
+      "Uploading directory to IPFS. This may take a while depending on file sizes.",
+    );
+    const uris = await storage.uploadBatch(files);
```

Comment:
Not for right now, but adding progress % here would be clutch

Reply:
Yea was thinking that - really wanted it but no upload progress for node :(